### PR TITLE
fix(RestAPI):500 Internal server error from releases API.

### LINKED ISSUE #1358: 500 Internal server error from releases API
[//]: # (Copyright Siemens AG, 2021. Part of the SW360 Portal Project)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

### Description

user are getting 500 Internal server error calling this endpoint
https://localhost:8080/resource/api/releases/b2a2f798ae274da0b711c8271d46431c/attachments
with response
{
    "timestamp": "2021-07-01T15:42:06.631444Z",
    "status": 500,
    "error": "Internal Server Error",
    "message": null
}
It started to behave correctly after adding one attachment.
Seems it has some problem when there is no attachment.

#### How to reproduce

Describe the bug and list the steps you used when the issue occurred.

### Versions

* Last commit id on master:
* Operating System (lsb_release -a):

### Logs

Any logs (if any) generated in

#### SW360 logs

Logs generated under /var/log/sw360/sw360.log

#### Tomcat logs

Logs generated under /var/log/tomcat/error.log

### PR DESCRIPTION
**Pull request by ravi110336**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Fix the 500 Internal server error from releases API while fetching the attachment.
> * Which issue is this pull request belonging to and how is it solving it? (*#1358*)
> * Did you add or update any new dependencies that are required for your change?

Issue: closes #1358

### How To Test?

- create a component via rest api.
http://localhost:8080/resource/api/components
- create a release via rest api.
http://localhost:8080/resource/api/releases
- fetch the attachment for the release via rest api. (Do not add any attachment manually or via rest api)
http://localhost:8080/resource/api/releases/2f5154c0201b4be8836ebd80feec2e70/attachments

> How should these changes be tested by the reviewer?
> Have you implemented any additional tests?

### Checklist
Must:
- [X] All related issues are referenced in commit messages and in PR

Signed-off-by: ravi110336 <kumar.ravindra@siemens.com>

### CODE REVIEW

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/attachment/Sw360AttachmentService.java [hunk 1]
```diff
@@ -240,10 +240,12 @@ public AttachmentContent getAttachmentContent(String id) {
 
     public Resources<Resource<Attachment>> getResourcesFromList(Set<Attachment> attachmentList) {
         final List<Resource<Attachment>> attachmentResources = new ArrayList<>();
-        for (final Attachment attachment : attachmentList) {
-            final Attachment embeddedAttachment = restControllerHelper.convertToEmbeddedAttachment(attachment);
-            final Resource<Attachment> attachmentResource = new Resource<>(embeddedAttachment);
-            attachmentResources.add(attachmentResource);
+        if (attachmentList != null && !attachmentList.isEmpty()) {
```
Comment (akapti) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/attachment/Sw360AttachmentService.java:243]: please  use `CommonUtils.isNotEmpty(attachmentList)`.

Comment (akapti) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/attachment/Sw360AttachmentService.java:243]: please  use `CommonUtils.isNotEmpty(attachmentList)`.

**Review by akapti — CHANGES_REQUESTED** [review]
kindly make the minor change.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/attachment/Sw360AttachmentService.java (hunk 1 above)
Comment (ravi110336) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/attachment/Sw360AttachmentService.java]: Thanks @akapti for the review, i have used the CommonUtils.isNotEmpty(attachmentList).

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/attachment/Sw360AttachmentService.java [hunk 2]
```diff
@@ -240,10 +240,12 @@ public AttachmentContent getAttachmentContent(String id) {
 
     public Resources<Resource<Attachment>> getResourcesFromList(Set<Attachment> attachmentList) {
         final List<Resource<Attachment>> attachmentResources = new ArrayList<>();
-        for (final Attachment attachment : attachmentList) {
-            final Attachment embeddedAttachment = restControllerHelper.convertToEmbeddedAttachment(attachment);
-            final Resource<Attachment> attachmentResource = new Resource<>(embeddedAttachment);
-            attachmentResources.add(attachmentResource);
+        if (attachmentList != null && CommonUtils.isNotEmpty(attachmentList)) {
```
Comment (akapti) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/attachment/Sw360AttachmentService.java:243]: Kindly remove the null check, `CommonUtils.isNotEmpty` handles the null case.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/attachment/Sw360AttachmentService.java (hunk 1 above)
Comment (akapti) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/attachment/Sw360AttachmentService.java]: @ravi110336 : Kindly remove the null check, `CommonUtils.isNotEmpty` handles the null case.

Comment (ravi110336) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/attachment/Sw360AttachmentService.java]: Thanks @akapti removed the extra null check.

**Review by akapti — APPROVED** [review]
code looks good, tested the changes, 
Issue has been fixed.

**Pull request merged**